File: 0150 evaluate-reverse-polish-notation/solution.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut st: Vec<i32> = Vec::new();

        for token in tokens {
            match token.as_str() {
                "+" => {
                    let a = st.pop().unwrap();
                    let b = st.pop().unwrap();
                    st.push(b + a);
                }
                "-" => {
                    let a = st.pop().unwrap();
                    let b = st.pop().unwrap();
                    st.push(b - a);
                }
                "*" => {
                    let a = st.pop().unwrap();
                    let b = st.pop().unwrap();
                    st.push(b * a);
                }
                "/" => {
                    let a = st.pop().unwrap();
                    let b = st.pop().unwrap();
                    st.push(b / a);
                }
                num => {
                    st.push(num.parse::<i32>().unwrap());
                }
            }
        }

        st.pop().unwrap()
    }
}

File: 0150 evaluate-reverse-polish-notation/solution.ts
function evalRPN(tokens: string[]): number {
    let st = [];
    for(const a of tokens) {
        if (a == "+") {
            st.push(st.pop()+st.pop());
        } else if (a=="-") {
            let o = st.pop();
            let t = st.pop();
            st.push(t-o);
        } else if (a=="*") {
            st.push(st.pop()*st.pop());
        } else if (a=="/") {
            let o = st.pop();
            let t = st.pop();
            st.push(Math.trunc(t/o));
        } else {
            st.push(Number(a));
        }
    }
    return st[0];
};
